src/openrct2/world/Footpath.cpp

```diff
@@ -881,59 +877,59 @@ static void loc_6A6D7E(
                 {
                     neighbour_list_push(
-                        neighbourList, 4, direction, tileElement->AsPath()->GetRideIndex(),
+                        neighbourList, 4, pos.direction, tileElement->AsPath()->GetRideIndex(),
                         tileElement->AsPath()->GetStationIndex());
                 }
                 else
                 {
                     if ((initialTileElement)->GetType() == TILE_ELEMENT_TYPE_PATH && initialTileElement->AsPath()->IsQueue())
                     {
-                        if (footpath_disconnect_queue_from_path(x, y, tileElement, 0))
+                        if (footpath_disconnect_queue_from_path(shiftedPos, tileElement, 0))
                         {
                             neighbour_list_push(
-                                neighbourList, 3, direction, tileElement->AsPath()->GetRideIndex(),
+                                neighbourList, 3, pos.direction, tileElement->AsPath()->GetRideIndex(),
                                 tileElement->AsPath()->GetStationIndex());
                         }
                     }
                 }
             }
             else
             {
-                neighbour_list_push(neighbourList, 2, direction, 255, 255);
+                neighbour_list_push(neighbourList, 2, pos.direction, 255, 255);
             }
         }
         else
         {
-            footpath_disconnect_queue_from_path(x, y, tileElement, 1 + ((flags >> 6) & 1));
-            tileElement->AsPath()->SetEdges(tileElement->AsPath()->GetEdges() | (1 << direction_reverse(direction)));
+            footpath_disconnect_queue_from_path(shiftedPos, tileElement, 1 + ((flags >> 6) & 1));
+            tileElement->AsPath()->SetEdges(tileElement->AsPath()->GetEdges() | (1 << direction_reverse(pos.direction)));
             if (tileElement->AsPath()->IsQueue())
             {
                 footpath_queue_chain_push(tileElement->AsPath()->GetRideIndex());
             }
         }
         if (!(flags & (GAME_COMMAND_FLAG_GHOST | GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED)))
         {
-            footpath_interrupt_peeps(x, y, tileElement->GetBaseZ());
+            footpath_interrupt_peeps({ shiftedPos, tileElement->GetBaseZ() });
         }
-        map_invalidate_element({ x, y }, tileElement);
+        map_invalidate_element(shiftedPos, tileElement);
     }
 
 loc_6A6FD2:
     if ((initialTileElement)->GetType() == TILE_ELEMENT_TYPE_PATH)
     {
         if (!query)
         {
-            initialTileElement->AsPath()->SetEdges(initialTileElement->AsPath()->GetEdges() | (1 << direction));
-            map_invalidate_element({ initialX, initialY }, initialTileElement);
+            initialTileElement->AsPath()->SetEdges(initialTileElement->AsPath()->GetEdges() | (1 << pos.direction));
+            map_invalidate_element(pos, initialTileElement);
         }
     }
 }
 
 static void loc_6A6C85(
```

Comment:
Future refactoring job would be to change this into a simple check that validates if the direction should be fully checked with `loc_6A6D7E` and move the calling of `loc_6A6D7E` into the parent function.

Reply:
I've noted on my personal notes, do you want me to add a `TODO` as well?

Reply:
Yeah you could add a todo as it won't be for a while